website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js

```diff
@@ -0,0 +1,63 @@
+odoo.define("website_sale_product_minimal_price.shop_min_price", function(require) {
+    "use strict";
+
+    const publicWidget = require("web.public.widget");
+    const core = require("web.core");
+    const field_utils = require("web.field_utils");
+
+    publicWidget.registry.WebsiteSaleProductMinimalPrice = publicWidget.Widget.extend({
+        selector: "#products_grid",
+        xmlDependencies: [
+            "/website_sale_product_minimal_price/static/src/xml/website_sale_product_minimal_price.xml",
+        ],
+
+        start: function() {
+            return Promise.all([
+                this._super.apply(this, arguments),
+                this.render_price(),
+            ]);
+        },
+        render_price: async function() {
+            const $products = $(".o_wsale_product_grid_wrapper");
+            const product_dic = {};
+            $products.each(function() {
+                product_dic[this.querySelector("a img").src.split("/")[6]] = this;
+            });
+            const product_ids = Object.keys(product_dic).map(Number);
+            const products_min_price = await this._rpc({
```

Comment:
Please @CarlosRoca13 as @Tardo comment, make sure that pricelist is in the context for get the correct price.